packages/amplify-category-geo/src/service-walkthroughs/deviceLocationTrackingWalkthrough.ts

```diff
@@ -0,0 +1,129 @@
+    minLength(1, nameValidationErrMsg),
+    maxLength(95, nameValidationErrMsg),
+    uniquenessValidation,
+  ]);
+  const trackerNameInput = await prompter.input(
+    'Provide a name for the device location tracker:',
+    { validate: validator, initial: `tracker${initialTrackerId}` },
+  );
+  return { name: trackerNameInput };
+};
+
+/**
+ * deviceLocationTrackerAccessWalkthrough
+ */
+export const deviceLocationTrackerAccessWalkthrough = async (
+  context: $TSContext,
+  parameters: Partial<DeviceLocationTrackingParameters>,
+): Promise<Partial<DeviceLocationTrackingParameters>> => {
+  const updatedParameters = { ...parameters };
+  updatedParameters.accessType = AccessType.CognitoGroups;
+
+  const resourceAccessParams = await resourceAccessWalkthrough(context, updatedParameters, ServiceName.DeviceLocationTracking);
+
+  const selectedGroups = [];
+
+  if (resourceAccessParams.accessType === AccessType.AuthorizedAndGuestUsers) {
+    selectedGroups.push('authenticated', 'guest');
+  } else if (resourceAccessParams.accessType === AccessType.AuthorizedUsers) {
+    selectedGroups.push('authenticated');
+  }
+
+  if (resourceAccessParams.groupPermissions && resourceAccessParams.groupPermissions.length > 0) {
+    selectedGroups.push(...resourceAccessParams.groupPermissions);
+  }
+
+  // Flow for asking what permissions to apply for each cognito role or group
+  const groupCrudPermissionsFlow = async (group: string, defaults: string[] = []): Promise<string[]> => {
+    const selectedCrudPermissions = await prompter.pick<'many', string>(
+      `What kind of access do you want for ${group} users? Select ALL that apply:`,
+      Object.keys(deviceLocationTrackingCrudPermissionsMap),
+      { returnSize: 'many', initial: byValues(defaults), pickAtLeast: 1 },
+    );
+    return selectedCrudPermissions;
+  };
+
+  const selectedGroupPermissions: Record<string, string[]> = {};
+
+  // Creates a mapping of authRole, unauthRole, and any selected groups to the selected crud permissions
+  for (const selectedUserPoolGroup of selectedGroups) {
+    const defaults = parameters?.roleAndGroupPermissionsMap?.[selectedUserPoolGroup] || [];
+
+    const selectedCrudPermissions = await groupCrudPermissionsFlow(selectedUserPoolGroup, defaults);
+
+    selectedGroupPermissions[selectedUserPoolGroup] = selectedCrudPermissions;
+  }
+
+  updatedParameters.groupPermissions = resourceAccessParams.groupPermissions;
+  updatedParameters.roleAndGroupPermissionsMap = selectedGroupPermissions;
+  return updatedParameters;
+};
+
+// const deviceLocationTrackerAdvancedWalkthrough = (): void => {
+
+// };
+
+// const deviceLocationTrackerGeofenceLinkingWalkthrough = async (context: $TSContext): Promise<> => {
+
+// };
+
+// const deviceLocationTrackerKMSSettingsWalkthrough = async (context: $TSContext): Promise<> => {
+
+// };
+
+// const deviceLocationTrackerFilteringMethodWalkthrough = async (context: $TSContext): Promise<> => {
+
+// };
```

Comment:
Remove these lines if not needed.

Reply:
These are stubs for the advanced workflow that we will work on in a separate PR

Reply:
please add them in separate PR then.